### PR TITLE
Upgrade compatibility and fix vulnerability

### PR DESCRIPTION
**Pull request by adojck**
Now in `resetStyles()` the default form styling is picked up from `newsletter2go_default.json` file and not `$_POST`

This fixes security vulnerability

Tested with Wordpress 6.5.2 and PHP 8.3

### CODE REVIEW

File: src/gui/N2Go_Gui.php [hunk 1]
```diff
@@ -86,12 +86,12 @@ public function adminOptions()
         }
 
         if ($_SERVER['REQUEST_METHOD'] == 'POST') {
-            $this->save_option('n2go_formUniqueCode', $_POST['formUniqueCode']);
+            $this->save_option('n2go_formUniqueCode', htmlspecialchars($_POST['formUniqueCode']));
 
-            $widgetStyleConfig = json_decode(stripcslashes($_POST['widgetStyleConfig']));
```
Comment (aahmadts) [inline src/gui/N2Go_Gui.php:91]: I see this was not used before but shouldn't we keep this and use it to avoid sql injections?

File: src/gui/N2Go_Gui.php [hunk 2]
```diff
@@ -136,8 +136,6 @@ public function adminOptions()
         if (strlen($formUniqueCode) > 0 && !isset($forms[$formUniqueCode])) {
             $this->save_option('n2go_formUniqueCode', null);
             $formUniqueCode = null;
-        } else {
-            $form = $forms[$formUniqueCode];
```
Comment (aahmadts) [inline src/gui/N2Go_Gui.php:140]: why is this removed. won't this break anything?

File: src/gui/N2Go_Gui.php [hunk 3]
```diff
@@ -295,7 +293,7 @@ private function disconnect()
      */
     function resetStyles()
     {
-        $style = $_POST['style'];
+        $style = file_get_contents(plugins_url('/lib/newsletter2go_default.json', __FILE__));
```
Comment (aahmadts) [inline src/gui/N2Go_Gui.php:296]: I am not sure if this will work. each form can be different and forms could have different input fields and customer html tags as well

File: src/gui/N2Go_Gui.php (hunk 1 above)
Comment (adojck) [reply src/gui/N2Go_Gui.php:91]: Not fully sure what you mean. You are correct about that it was not used, it was intended to check if `n2go_widgetStyleConfig` option key needs update. I agree that this line 93:
```
$this->save_option('n2go_widgetStyleConfig', $_POST['widgetStyleConfig']);
```
Is not the best, as we pass `$_POST` data directly to database, but I was not sure how to make it more safe. Note that in database we save this as plain JSON in a text field (`longtext`)

I think I could run `json_decode` and later `json_encode` to strip out stuff which should not be there. What do you think?

File: src/gui/N2Go_Gui.php (hunk 2 above)
Comment (adojck) [reply src/gui/N2Go_Gui.php:140]: This was causing a PHP warning like this:
```
PHP Warning:  Trying to access array offset on false in /www/wp-content/plugins/newsletter2go-wordpress-plugin/src/gui/N2Go_Gui.php on line 140
```

`$forms` is not an array, but just bool false as there are no forms at this point

Was not sure what purpose of this was, as `$form` is not used anywhere, and after removing it I did not encounter any issues

File: src/gui/N2Go_Gui.php (hunk 3 above)
Comment (adojck) [reply src/gui/N2Go_Gui.php:296]: I tested, it works. What it does, it just loads default styling for the form from `newsletter2go_default.json` file, instead of loading in runtime from Javascript file and passing via POST method back to WP.

In both cases it always overwrites custom styling with default, so no actual change is happening, just it happens in a different way.

File: src/gui/N2Go_Gui.php (hunk 2 above)
Comment (aahmadts) [reply src/gui/N2Go_Gui.php:140]: `$forms` is an array and will be false if the authKey is not set in the db. 
removing this will fore sure break something
check  https://github.com/newsletter2go/newsletter2go-wordpress-plugin/blob/82fbcef114de09de9b0c87107dfcc943be83494f/src/gui/N2Go_Gui.php#L171

File: src/gui/N2Go_Gui.php (hunk 1 above)
Comment (aahmadts) [reply src/gui/N2Go_Gui.php:91]: ok I get now. then it can stay as now since the input is done from admin backend anyways.

File: src/gui/N2Go_Gui.php (hunk 3 above)
Comment (aahmadts) [reply src/gui/N2Go_Gui.php:296]: Not sure. I still need to check

File: src/gui/N2Go_Gui.php (hunk 2 above)
Comment (adojck) [reply src/gui/N2Go_Gui.php:140]: The thing is that it does not break anything, just emits warning. Note that `$form` is not used anywhere anyways. It's ok if `$forms` is false, as we care only when it's an array, and even then we don't do anything with `$form`

Comment (aahmadts) [reply src/gui/N2Go_Gui.php:140]: you are right. `$form` is not used anywhere

File: src/gui/N2Go_Gui.php (hunk 3 above)
Comment (aahmadts) [reply src/gui/N2Go_Gui.php:296]: right! better like that

**Review by aahmadts — APPROVED** [review]
